common/network-common/src/main/java/org/apache/spark/network/sasl/SaslClientBootstrap.java

```diff
@@ -65,9 +67,18 @@ public void doBootstrap(TransportClient client, Channel channel) {
         SaslMessage msg = new SaslMessage(appId, payload);
         ByteBuf buf = Unpooled.buffer(msg.encodedLength() + (int) msg.body().size());
         msg.encode(buf);
+        ByteBuffer response;
         buf.writeBytes(msg.body().nioByteBuffer());
-
-        ByteBuffer response = client.sendRpcSync(buf.nioBuffer(), conf.authRTTimeoutMs());
+        try {
+          response = client.sendRpcSync(buf.nioBuffer(), conf.authRTTimeoutMs());
+        } catch (RuntimeException ex) {
+          // We know it is a Sasl timeout here if it is a TimeoutException.
+          if (ex.getCause() instanceof TimeoutException) {
+            throw Throwables.propagate(new TransportClient.SaslTimeoutException(ex.getCause()));
```

Comment:
Why `Throwables.propagate` ? Simply create and throw the new `SaslTimeoutException` directly ?